### PR TITLE
implementing functionality for node changing

### PR DESCRIPTION
**Pull request by andrewkryshtal**

### CODE REVIEW

File: src/components/styled-radio-buttons-group/styled-radio-buttons-group.tsx [hunk 1]
```diff
@@ -31,7 +32,7 @@ export const StyledRadioButtonsGroup = <T extends string>({ value, buttons, onCh
         ...radioButton,
         id: radioButton.value,
         labelStyle: styles.label,
-        containerStyle: styles.itemContainer,
+        containerStyle: [styles.itemContainer],
```
Comment (unixvb) [inline src/components/styled-radio-buttons-group/styled-radio-buttons-group.tsx:35]: remove array

File: src/components/styled-radio-buttons-group/styled-radio-buttons-group.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ interface Props<T extends string> extends RadioButtonsGroupProps<T> {
   onChange: EventFn<T>;
 }
 
+/*** @deprecated */
```
Comment (unixvb) [inline src/components/styled-radio-buttons-group/styled-radio-buttons-group.tsx:24]: remove this

File: src/enums/network.enum.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export enum NetworkEnum {
+  TEMPLE_DEFAULT = 'TEMPLE_DEFAULT',
```
Comment (unixvb) [inline src/enums/network.enum.ts:2]: add giganode

File: src/deprecated/deprecated-styled-radio-buttons-group/deprecated-styled-radio-buttons-group.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+export interface DeprecatedRadioButtonsGroupProps<T extends string> {
+  buttons: DeprecatedRadioButton<T>[];
+}
+
+interface Props<T extends string> extends DeprecatedRadioButtonsGroupProps<T> {
+  value: T;
+  onChange: EventFn<T>;
+}
+
+/*** @deprecated */
```
Comment (unixvb) [inline src/deprecated/deprecated-styled-radio-buttons-group/deprecated-styled-radio-buttons-group.tsx:24]: delete depracated

File: src/utils/network/networks.ts [hunk 5]
```diff
@@ -1,13 +1,41 @@
...
-  id: 'mainnet',
-  name: 'tezosMainnet',
-  nameI18nKey: 'tezosMainnet',
-  description: 'Tezos mainnet',
-  lambdaContract: 'KT1CPuTzwC7h7uLXd5WQmpMFso1HxrLBUtpE',
-  type: 'main',
-  rpcBaseURL: 'https://mainnet-node.madfish.solutions',
-  color: '#83b300',
-  disabled: false
+export const NETWORKS: NetworkRecord = {
```
Comment (unixvb) [inline src/utils/network/networks.ts:4]: rename everything to `Rpc`s and remove unused properties

File: src/hooks/use-radio-buttons-group.hook.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { RadioButtonProps } from 'react-native-radio-buttons-group/lib/index';
+
+import { EventFn } from '../config/general';
+import { isDefined } from '../utils/is-defined';
+
+interface Props<T extends string> {
+  onChange: EventFn<T>;
+}
+
+export const useRadioButtonsGroupHook = <T extends string>({ onChange }: Props<T>) => {
```
Comment (unixvb) [inline src/hooks/use-radio-buttons-group.hook.ts:10]: not used

File: src/enums/network.enum.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export enum NetworkEnum {
```
Comment (unixvb) [inline src/enums/network.enum.ts:1]: rename to `Rpc` naming

File: src/interfaces/network.interface.ts [hunk 8]
```diff
@@ -1,11 +1,14 @@
+import { NetworkEnum } from '../enums/network.enum';
+
 export interface NetworkInterface {
```
Comment (unixvb) [inline src/interfaces/network.interface.ts:3]: rename to `Rpc` naming

File: src/interfaces/network.interface.ts [hunk 9]
```diff
@@ -1,11 +1,14 @@
...
   description: string;
   lambdaContract: string;
-  type: string;
   rpcBaseURL: string;
-  color: string;
-  disabled: boolean;
-  nameI18nKey?: string;
 }
+
+export type NetworkRecord = {
```
Comment (unixvb) [inline src/interfaces/network.interface.ts:12]: rename to `Rpc` naming

File: src/modals/confirmation-modal/operations-confirmation/hooks/use-estimations.hook.ts [hunk 10]
```diff
@@ -13,10 +13,10 @@ export const useEstimations = (sender: WalletAccountInterface, opParams: ParamsW
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    const subscription = from(
-      tezos$.getValue().estimate.batch(opParams.map(param => ({ ...param, source: sender.publicKeyHash })))
-    )
+    const subscription = of(null)
```
Comment (unixvb) [inline src/modals/confirmation-modal/operations-confirmation/hooks/use-estimations.hook.ts:16]: don't like this.  
rewrite with
```
tezos$.getValue().estimate.batch(...)
```

File: src/screens/node-settings/node-settings.tsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  useNavigationSetOptions(
+    {
+      headerLeft: () => <HeaderBackButton />
+    },
+    []
+  );
+
+  return (
+    <ScreenContainer>
+      <StyledRadioButtonsGroup onChange={onChangeNodeHandler} value={node as NetworkEnum} buttons={nodesButtons} />
```
Comment (unixvb) [inline src/screens/node-settings/node-settings.tsx:36]: try t remove this assertion. (remove `| string` type from `useState`)

File: src/screens/node-settings/node-settings.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+export const NodeSettings = () => {
+  const [node, setNode] = useState<NetworkEnum | string>(NetworkEnum.TEMPLE_DEFAULT);
+
+  const onChangeNodeHandler = (value: NetworkEnum) => {
+    AsyncStorage.setItem('nodeInstance', value).then(() => updateCurrentNetwork(NETWORKS[value]));
+  };
+
+  useEffect(() => {
+    AsyncStorage.getItem('nodeInstance').then(data => setNode(data ?? NetworkEnum.TEMPLE_DEFAULT));
+  }, []);
```
Comment (unixvb) [inline src/screens/node-settings/node-settings.tsx:25]: short return

File: src/screens/node-settings/node-settings.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+import { useNavigationSetOptions } from '../../components/header/use-navigation-set-options.hook';
+import { ScreenContainer } from '../../components/screen-container/screen-container';
+import { StyledRadioButtonsGroup } from '../../components/styled-radio-buttons-group/styled-radio-buttons-group';
+import { NetworkEnum } from '../../enums/network.enum';
+import { updateCurrentNetwork } from '../../utils/network/network.util';
+import { NETWORKS } from '../../utils/network/networks';
+
+const nodesButtons = (Object.keys(NETWORKS) as Array<NetworkEnum>).map((item: NetworkEnum) => {
+  return { ...NETWORKS[item], value: NETWORKS[item].id };
+});
```
Comment (unixvb) [inline src/screens/node-settings/node-settings.tsx:14]: short return

File: src/screens/node-settings/node-settings.tsx [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+import { HeaderBackButton } from '../../components/header/header-back-button/header-back-button';
+import { useNavigationSetOptions } from '../../components/header/use-navigation-set-options.hook';
+import { ScreenContainer } from '../../components/screen-container/screen-container';
+import { StyledRadioButtonsGroup } from '../../components/styled-radio-buttons-group/styled-radio-buttons-group';
+import { NetworkEnum } from '../../enums/network.enum';
+import { updateCurrentNetwork } from '../../utils/network/network.util';
+import { NETWORKS } from '../../utils/network/networks';
+
+const nodesButtons = (Object.keys(NETWORKS) as Array<NetworkEnum>).map((item: NetworkEnum) => {
```
Comment (unixvb) [inline src/screens/node-settings/node-settings.tsx:12]: could you omit assertion?

File: src/utils/network/network.util.ts [hunk 15]
```diff
@@ -23,7 +24,13 @@ export const createTezosToolkit = (rpc: string) => {
 
 export const getFastRpcClient = memoize((rpc: string) => new FastRpcClient(rpc));
 
-const currentNetwork$ = new BehaviorSubject(MAINNET_NETWORK);
+export const currentNetwork$ = new BehaviorSubject(NETWORKS.TEMPLE_DEFAULT);
```
Comment (unixvb) [inline src/utils/network/network.util.ts:27]: rename to `Rpc` naming

File: src/utils/network/network.util.ts [hunk 16]
```diff
@@ -23,7 +24,13 @@ export const createTezosToolkit = (rpc: string) => {
...
 export const currentNetworkId$ = currentNetwork$.pipe(map(({ id }) => id));
 export const currentNetworkRpc$ = currentNetwork$.pipe(map(({ rpcBaseURL }) => rpcBaseURL));
-export const tezos$ = new BehaviorSubject(createTezosToolkit(MAINNET_NETWORK.rpcBaseURL));
+// export const tezos$ = new BehaviorSubject(createTezosToolkit(MAINNET_NETWORK.rpcBaseURL));
```
Comment (unixvb) [inline src/utils/network/network.util.ts:30]: remove commented line

File: src/utils/network/network.util.ts [hunk 17]
```diff
@@ -23,7 +24,13 @@ export const createTezosToolkit = (rpc: string) => {
...
+export const tezos$ = currentNetworkRpc$.pipe(map(rpcBaseUrl => createTezosToolkit(rpcBaseUrl)));
+
+export const updateCurrentNetwork = (networkInterface: NetworkInterface) => {
+  currentNetwork$.next(networkInterface);
+  console.log('network changed ', networkInterface);
+};
```
Comment (unixvb) [inline src/utils/network/network.util.ts:36]: `networkInterface`?  
remove console.log   
short return

File: src/interfaces/network.interface.ts [hunk 18]
```diff
@@ -1,11 +1,14 @@
+import { NetworkEnum } from '../enums/network.enum';
+
 export interface NetworkInterface {
-  id: string;
+  id: NetworkEnum;
+  label: string;
   name: string;
   description: string;
   lambdaContract: string;
```
Comment (unixvb) [inline src/interfaces/network.interface.ts:8]: no need in `lambdaContract`

File: src/enums/network.enum.ts (hunk 3 above)
Comment (andrewkryshtal) [reply src/enums/network.enum.ts:2]: already added on last enum

File: src/screens/node-settings/node-settings.tsx (hunk 11 above)
Comment (andrewkryshtal) [reply src/screens/node-settings/node-settings.tsx:36]: fixed, but added assertion on incoming data in `getItem` method

Comment (andrewkryshtal) [reply src/screens/node-settings/node-settings.tsx]: i put assertion only in one place - setNode, defining data as `RpcEnum`. Can't figure how to fix it

File: src/modals/confirmation-modal/operations-confirmation/hooks/use-estimations.hook.ts (hunk 10 above)
Comment (andrewkryshtal) [reply src/modals/confirmation-modal/operations-confirmation/hooks/use-estimations.hook.ts]: we replaced old `tezos$` method with observable so i decided to use just `tezos$`. in current code `tezos$` doesn't have method `getValue()`

File: src/interfaces/network.interface.ts [hunk 19]
```diff
@@ -1,11 +1,13 @@
...
+import { RpcEnum } from '../enums/network.enum';
+
+export interface RpcInterface {
+  id: RpcEnum;
+  label: string;
   name: string;
   description: string;
-  lambdaContract: string;
-  type: string;
   rpcBaseURL: string;
```
Comment (unixvb) [inline src/interfaces/network.interface.ts:8]: baseURL

**Review by keshan3262 — APPROVED** [review]

File: src/enums/network.enum.ts [hunk 20]
```diff
@@ -0,0 +1,6 @@
+export enum RpcEnum {
```
Comment (unixvb) [inline src/enums/network.enum.ts:1]: file name

File: src/interfaces/network.interface.ts [hunk 21]
```diff
@@ -1,11 +1,9 @@
-export interface NetworkInterface {
-  id: string;
+import { RpcEnum } from '../enums/network.enum';
+
+export interface RpcInterface {
```
Comment (unixvb) [inline src/interfaces/network.interface.ts:3]: file name

File: src/modals/confirmation-modal/operations-confirmation/hooks/use-estimations.hook.ts [hunk 22]
```diff
@@ -26,7 +26,8 @@ export const useEstimations = (sender: WalletAccountInterface, opParams: ParamsW
             minimalFeePerStorageByteMutez
           }))
         ),
-        catchError(() => {
+        catchError(err => {
+          console.log({ err });
```
Comment (unixvb) [inline src/modals/confirmation-modal/operations-confirmation/hooks/use-estimations.hook.ts:30]: remove this

File: src/navigator/enums/screens.enum.ts [hunk 23]
```diff
@@ -74,5 +76,6 @@ export const settingsStackScreens = [
   ScreensEnum.ManageAccounts,
   ScreensEnum.About,
   ScreensEnum.DAppsSettings,
+  ScreensEnum.SecureSettings,
   ScreensEnum.SecureSettings
```
Comment (unixvb) [inline src/navigator/enums/screens.enum.ts:80]: `SecureSettings` twice

File: src/utils/network/rpc-record.ts [hunk 24]
```diff
@@ -0,0 +1,33 @@
+import { RpcEnum } from '../../enums/network.enum';
+import { RpcInterface } from '../../interfaces/network.interface';
+
+export const RPC: RpcInterface[] = [
```
Comment (unixvb) [inline src/utils/network/rpc-record.ts:4]: maybe `RpcArray`?  
And file name is incorrect, it's not record any more

File: src/shelter/use-shelter.hook.ts [hunk 25]
```diff
@@ -57,22 +57,17 @@ export const useShelter = () => {
...
           if (publicData !== undefined) {
             dispatch(setSelectedAccountAction(publicData.publicKeyHash));
             dispatch(addHdAccountAction(publicData));
             showSuccessToast({ description: 'Account Imported!' });
             goBack();
+          } else {
+            showErrorToast({
+              title: 'Failed to import account.',
+              description: 'This may happen because provided Key is invalid.'
+            });
```
Comment (unixvb) [inline src/shelter/use-shelter.hook.ts:70]: revert this changes, not related to this pr

**Review by unixvb — APPROVED** [review]
